src/d/actor/d_a_ff.cpp

```diff
@@ -1,55 +1,363 @@
+        f32 fVar1 = 0.5f - param_2->z * -1.0f;
+        if (fVar1 < 0.0f) {
+            fVar1 = 0.0f;
+        }
+        if ((u32)(fVar1 * 33553002.0f) > local_28) {
+            i_this->mbNotVisibleZ = 1;
+        } else {
+            i_this->mbNotVisibleZ = 0;
+        }
+    }
+    return;
 }
 
 /* 000002EC-0000037C       .text daFf_Draw__FP8ff_class */
-static BOOL daFf_Draw(ff_class*) {
-    /* Nonmatching */
+static BOOL daFf_Draw(ff_class* i_this) {
+    cXyz cStack_18;
+
+    mDoLib_project(&i_this->current.pos, &cStack_18);
+    z_check(i_this, &cStack_18);
+    if (i_this->mbNotVisibleZ != 0) {
+        cLib_addCalc0(&i_this->mGlowScale, 1.0f, 0.33333f);
+    } else {
+        cLib_addCalc2(&i_this->mGlowScale, 1.0f, 1.0f, 0.3333f);
+    }
+    fire_fly_draw(i_this);
+    return TRUE;
 }
 
 /* 0000037C-00000D18       .text fire_fly_move__FP8ff_class */
-void fire_fly_move(ff_class*) {
-    /* Nonmatching */
+static void fire_fly_move(ff_class* i_this) {
+    /* Nonmatching - regalloc */
+    s8 cVar1;
+    f32 fVar2;
+    f32 fVar3;
+    f32 dVar7;
+    f32 dVar8;
+    f32 dVar9;
+    f32 fVar10;
+    cXyz local_cc;
+
+    daPy_py_c* player = (daPy_py_c*)dComIfGp_getPlayer(0);
+    dBgS_GndChk chk;
+    i_this->mLiveTimer++;
+    if (i_this->mTimers[2] == 0) {
+        i_this->mTimers[1] = cM_rndF(50.0f) + 40.0f;
+        i_this->mTimers[2] = cM_rndF(200.0f) + 100.0f;
+    }
+    if (i_this->mTimers[1] == 0) {
+        i_this->mScaleTarget = cM_ssin(i_this->mLiveTimer * 1000) * 0.15f * 0.25f + 0.225f;
+    } else {
+        i_this->mScaleTarget = 0.0f;
+    }
+    cLib_addCalc2(&i_this->mScale, i_this->mScaleTarget, 0.1f, 0.05f);
+    cVar1 = i_this->mState;
+    if (cVar1 == 2) {
+        if (cVar1 < 2) {
+            if (cVar1 == 0) {
+                chk.GetPointP()->set(i_this->current.pos.x, i_this->current.pos.z + 250.0f,
+                                     i_this->current.pos.y);
+                fVar10 = dComIfG_Bgsp()->GroundCross(&chk);
+                i_this->mGroundY = fVar10 + 12.5f;
+                if (i_this->mbNoUseGroundY == 0) {
+                    i_this->current.pos.y = i_this->mGroundY;
+                }
+                i_this->mHomePos = i_this->current.pos;
+                i_this->mState++;
+            }
+            cLib_addCalc2(&i_this->current.pos.x, i_this->mHomePos.x, 0.1f,
+                          std::abs(i_this->speed.x));
+            cLib_addCalc2(&i_this->current.pos.y, i_this->mHomePos.y, 0.1f,
+                          std::abs(i_this->speed.y));
+            cLib_addCalc2(&i_this->current.pos.z, i_this->mHomePos.z, 0.1f,
+                          std::abs(i_this->speed.z));
+            fVar10 = player->current.pos.x - i_this->current.pos.x;
+            fVar2 = player->current.pos.y - i_this->current.pos.y;
+            fVar3 = player->current.pos.z - i_this->current.pos.z;
+            fVar10 = std::sqrtf(fVar3 * fVar3 + fVar10 * fVar10 + fVar2 * fVar2);
```

Comment:
There is a `cXyz.abs` function that performs this sort of distance calculation between two vectors, that might be what was used here.

Reply:
C++-style classes usually use member methods like `cXyz::abs`, while C-style classes like this one usually don't and instead prefer to manually do the math.